README.md

```diff
@@ -1,44 +1,82 @@
 
-The build is minified and the filenames include the hashes.<br>
-Your app is ready to be deployed!
+## Requirements
 
-See the section about [deployment](https://facebook.github.io/create-react-app/docs/deployment) for more information.
+Each of the waves includes a number of questions. Though you should write your own code, feel free to collaborate with other students as you work on these questions.
 
-### `npm run eject`
+### Wave 0: Reading
 
-**Note: this is a one-way operation. Once you `eject`, you can’t go back!**
+We have already implemented some parts of this application:
 
-If you aren’t satisfied with the build tool and configuration choices, you can `eject` at any time. This command will remove the single build dependency from your project.
+- Code to load JSON pet data in `App.js`
+- 3 functional stateless components: `PetList`, `PetCard` and `PetDetails`
+- CSS for all components. You shouldn't need to write any CSS for this project.
+  - All of our CSS uses [BEM naming](http://getbem.com/naming/)
 
-Instead, it will copy all the configuration files and the transitive dependencies (Webpack, Babel, ESLint, etc) right into your project so you have full control over them. All of the commands except `eject` will still work, but they will point to the copied scripts so you can tweak them. At this point you’re on your own.
+Before you start writing code, read through what's already here and make sure you understand it by answering the following questions:
 
-You don’t have to ever use `eject`. The curated feature set is suitable for small and middle deployments, and you shouldn’t feel obligated to use this feature. However we understand that this tool wouldn’t be useful if you couldn’t customize it when you are ready for it.
+- How do the components relate to each other? Draw a diagram.
+- How does data get from `App.js` to `PetCard.js`?
 
-## Learn More
+### Wave 1: Viewing Details
 
-You can learn more in the [Create React App documentation](https://facebook.github.io/create-react-app/docs/getting-started).
+When the user clicks the `select` button the app should present the user with additional details on the selected pet.  Read through the provided `propTypes` attribute of the given components, **what information does that provide**?
 
-To learn React, check out the [React documentation](https://reactjs.org/).
+**Questions:**
+- How will you track which, if any Pet is the currently selected Pet?  Which component will keep the state?
+- Will you need to switch a functional component to a classical component?
+- What events should you listen for?
+- Draw a diagram of the flow of rendering and callbacks in your app so far, similar to the one we drew in class.
+
+### Wave 2: Removing Pets
+
+When the user clicks the `X` button on a PetCard, that animal should be removed from the application, including if that animal is the currently selected pet.  You should ensure that pets do not return when subsequent pets are removed.  
```

Comment:
What do you mean by the last line "You should ensure that pets do not return when subsequent pets are removed.